core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/GuardrailsSupport.java

```diff
@@ -140,15 +143,26 @@ private static InputGuardRailsResult invokeInputGuardRails(AiServiceMethodCreate
             }
         }
 
+        InputGuardrailResult result = InputGuardrailResult.success();
+
         for (Class<? extends InputGuardrail> bean : classes) {
-            try {
-                CDI.current().select(bean).get().validate(params);
-            } catch (Exception e) {
-                return new InputGuardRailsResult(false, bean, e);
-            }
+            result = compose(result, CDI.current().select(bean).get().validate(params).validatedBy(bean));
```

Comment:
Not using exceptions as a result for a validation, allow to easily compose more validations. This implies a change in the behaviour: now all the validations are performed regardless of the result instead of stopping at the first failure. I see this as an improvement since, in general, when performing a set of validations you want to run all of them and have a complete list of everything that went wrong instead of stopping at the first one. However I don't know if this reasoning applies well also to this specific context, but if not it will be easy to revert it and stop at the first failure. Even better we could leave this decision to users and allow them to return a `InputGuardrailResult` that can be _terminal_ (no other validation will be executed) or not.

Reply:
Maybe failure needs a 'fatal' flag that gives us the possibility to do both.